src/system_error.cr

```diff
@@ -43,7 +43,7 @@ module SystemError
   end
 
   # The original system error wrapped by this exception
-  getter os_error : Errno | WinError | Nil
+  getter os_error : Errno | WinError | WasiError | Nil
```

Comment:
💭  Maybe at this point we should create an alias for this union (optional in this PR!)

Reply:
I don't like much the fact that we have this union, since a program will only ever ever have one of these kinds of errors. It doesn't make sense to have all of those defined, they should be guarded by flags or moved to the Crystal::System, I don't know.